portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -0,0 +1,45 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/home")
+public class HomeServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    PrintWriter out = response.getWriter();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Responds with login status of user
+    boolean loggedIn = userService.isUserLoggedIn();
+    String url;
+    if (loggedIn) {
+      String logoutUrl = userService.createLogoutURL("/");
+      url = logoutUrl;
+    } else {
+      String loginUrl = userService.createLoginURL("/");
+      url = loginUrl;
+    }
+
+    String json = "{";
+    json += "\"isLoggedIn\": ";
+    json += loggedIn;
+    json += ", ";
+    json += "\"url\": ";
+    json += "\"" + url + "\"";
+    json += "}";
+    out.println(json);
```

Comment:
to avoid creating your own json you might consider creating a class and then using gson's automatic conversion ability.

Reply:
Is it okay if I leave it like this? The tutorial does it both ways, so I also wanted to practice doing it both ways in my code. Ie In DataServlet.java I did what you suggested above by creating a CommentResponse object which was applied to Gson's conversion method. 
However, I do see how having my code do the same thing with different implementations in different places could make it harder to read, so if you still want me to change this part as well for consistency, please let me know!

Reply:
Since you were doing it to learn then it is fine to keep.  I was just pointing out that that for things that are error prone and a lot of boilerplate that it is usually better to rely on automatic conversions as it will save you time in debugging because you will inevitably get it wrong and updating it in the future.

Reply:
Once you have read this you can resolve the conversation.